File: Week 5/A/ccc96s3.cpp
#include <bits/stdc++.h>
using namespace std;
void recur(string s, int ones, int length) {
    // current string, 1s remaining, # of characters remaining
    if(length == 0) cout << s << "\n"; // base case 
    if(ones > 0) recur(s+'1',ones-1,length-1);
    if(length > ones) recur(s+'0',ones,length-1);
}
int main() {
    int t; cin >> t;
    while(t--) {
        int n, k; cin >> n >> k;
        string s="";
        cout << "The bit patterns are\n";
        recur(s,k,n);
        cout << "\n"; 
    } 
}

File: 2020-2021/Week 5/A/ccc96s3.py
def recur(s, ones, length): 
    # current string, 1s left, remaining characters
    if(length==0):print(s) # base case
    if(ones > 0):recur(s+'1',ones-1,length-1) 
    if(length > ones):recur(s+'0',ones,length-1)

for i in range(int(input())):
    n,k=map(int,input().split())
    print("The bit patterns are")
    recur("",k,n)
    print()
